### DISCUSSION TITLE
how to map the `focus-visible:` modifier to `[data-focus-visible]`?

### DISCUSSION THREAD

**Discussion by stefanprobst** [tailwindlabs/tailwindcss#11679] [Help] [answered]
i would want to e.g. write

`focus-visible:text-primary-800`,

which should generate

`.focus-visible\:text-primary-800[data-focus-visible] { /* ... */ }`.

is that possible?

**Comment by wongjn** [comment, marked as answer]
Yep! Override the core variant via a [Tailwind plugin](https://tailwindcss.com/docs/plugins#static-variants):

```js
require('tailwindcss/plugin')(({ addVariant }) => {
  addVariant('focus-visible', '&[data-focus-visible]')
})
```

Live example: https://play.tailwindcss.com/6HcERH0lsF?file=config

**Reply by stefanprobst** [reply]
nice, thanks!
